Library/Homebrew/extend/ARGV.rb

```diff
@@ -13,7 +13,7 @@ def flags_only
 
   def formulae
     require "formula"
-    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory(name, spec) }
+    @formulae ||= (downcased_unique_named - casks).map { |name| Formulary.factory_with_priority(name, spec) }
```

Comment:
You should not prioritize bottle paths. The bottle formula is auto resolved based on what formula is used to build them. Related code is [here](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/bottles.rb#L45-L55).

What do you mean by formula path like `ack.rb`? A full formula file path? If so, it has nothing to do with prioritize as well.